react/features/base/react/components/web/MultiSelectAutocomplete.js

```diff
@@ -0,0 +1,316 @@
+import React, { Component } from 'react';
+import { MultiSelectStateless } from '@atlaskit/multi-select';
+import AKInlineDialog from '@atlaskit/inline-dialog';
+import Spinner from '@atlaskit/spinner';
+
+import InlineDialogFailure from './InlineDialogFailure';
+
+/**
+ * A MultiSelect that is also auto-completing.
+ */
+class MultiSelectAutocomplete extends Component {
+
+    /**
+     * {@code MultiSelectAutocomplete} component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The default value of the selected item.
+         */
+        defaultValue: React.PropTypes.array,
+
+        /**
+         * Indicates if the component is disabled.
+         */
+        isDisabled: React.PropTypes.bool,
+
+        /**
+         * The text to show when no matches are found.
+         */
+        noMatchesFound: React.PropTypes.string,
+
+        /**
+         * The function called when the selection changes.
+         */
+        onSelectionChange: React.PropTypes.func,
+
+        /**
+         * The placeholder text of the input component.
+         */
+        placeholder: React.PropTypes.string,
+
+        /**
+         * The service providing the search.
+         */
+        resourceClient: React.PropTypes.shape({
+            makeQuery: React.PropTypes.func,
+            parseResults: React.PropTypes.func
+        }).isRequired,
+
+        /**
+         * Indicates if the component should fit the container.
+         */
+        shouldFitContainer: React.PropTypes.bool,
+
+        /**
+         * Indicates if we should focus.
+         */
+        shouldFocus: React.PropTypes.bool
+    };
+
+    /**
+     * Initializes a new {@code MultiSelectAutocomplete} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        const defaultValue = this.props.defaultValue || [];
+
+        this.state = {
+            /**
+             * Indicates if the dropdown is open.
+             */
+            isOpen: false,
+
+            /**
+             * The filter value.
```

Comment:
Could you think of some more words here as a favor to me? Maybe what the filter value is used for?